apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -1,277 +1,328 @@
         });
+    }
+}
+
+function createCopyHandler(index: number) {
+    return () => {
+        copySchedule(index);
     };
+}
 
-    componentWillUnmount() {
-        AppStore.removeListener('addedCoursesChange', this.loadCourses);
-        AppStore.removeListener('customEventsChange', this.loadCustomEvents);
-        AppStore.removeListener('currentScheduleIndexChange', this.loadCourses);
-        AppStore.removeListener('currentScheduleIndexChange', this.loadCustomEvents);
-        AppStore.removeListener('scheduleNamesChange', this.loadScheduleNames);
-        AppStore.removeListener('scheduleNotesChange', this.loadScheduleNote);
-    }
+function SkeletonSchedule() {
+    const [skeletonSchedule, setSkeletonSchedule] = useState(AppStore.getSkeletonSchedule());
+
+    useEffect(() => {
+        const updateSkeletonSchedule = () => {
+            setSkeletonSchedule(AppStore.getSkeletonSchedule());
+        };
+
+        AppStore.on('skeletonScheduleChange', updateSkeletonSchedule);
 
-    loadCourses = () => {
-        const currentCourses = AppStore.schedule.getCurrentCourses();
-        const formattedCourses: CourseWithTerm[] = [];
-
-        for (const course of currentCourses) {
-            let formattedCourse: CourseWithTerm | undefined = formattedCourses.find(
-                (needleCourse) =>
-                    needleCourse.courseNumber === course.courseNumber && needleCourse.deptCode === course.deptCode
-            );
-
-            if (formattedCourse) {
-                formattedCourse.sections.push({
-                    ...course.section,
-                });
-            } else {
-                formattedCourse = {
-                    term: course.term,
-                    deptCode: course.deptCode,
-                    courseComment: course.courseComment,
-                    prerequisiteLink: course.prerequisiteLink,
-                    courseNumber: course.courseNumber,
-                    courseTitle: course.courseTitle,
-                    sections: [
-                        {
-                            ...course.section,
-                        },
-                    ],
-                };
-                formattedCourses.push(formattedCourse);
+        return () => {
+            AppStore.off('skeletonScheduleChange', updateSkeletonSchedule);
+        };
+    }, []);
+
+    const sectionsByTerm: [string, string[]][] = useMemo(() => {
+        const result = skeletonSchedule.courses.reduce((accumulated, course) => {
+            accumulated[course.term] ??= [];
+            accumulated[course.term].push(course.sectionCode);
+            return accumulated;
+        }, {} as Record<string, string[]>);
+
+        return Object.entries(result);
+    }, [skeletonSchedule.courses]);
+
+    return (
+        <>
+            {
+                // Sections organized under terms, in case the schedule contains multiple terms
+                sectionsByTerm.map(([term, sections], index) => (
```

Comment:
Sorry about all the styling nitpicks, but can we add a little padding here:

<img width="636" alt="Screenshot 2023-09-06 at 1 39 25 PM" src="https://github.com/icssc/AntAlmanac/assets/100006999/f7c690a9-bdb4-4d7c-95df-efdca3e1e5a3">

Reply:
Oh, I meant to wrap a padded box around the entire thing, but I forgot.